src/bin/book-test-instance.py

```diff
@@ -382,23 +382,21 @@ def vprint(*args):
         raise Exception('No pull request found, aborting.')
 
     # Fetch issue details from Github PR
-    try:
-        issue = get_jira_issue(pr=pr)
-    except Exception as e:
-        vprint(e)
-        issue = None
+    issue = get_jira_issue(pr=pr)
 
-    # Define instance
+    # If not a ticket PR just get an availiable instance
+    # Reverse the order to avoid race condition
     if not issue:
-        raise Exception('No corresponding issue found, booking will not be executed.')
-
-    if issue['test_instance']:
-        instance = issue['test_instance']
-        vprint('Issue is already deployed on {0}, reusing.'.format(instance))
+        instance = get_available_instance(reverse=True)
```

Comment:
Good catch that there could be a race condition when multiple PRs request an instance. But wouldn't this only prevent the race condition if 1 PR has no issue and the other does have one? If both have no issue or both have an issue it's still equally likely.

Reply:
Randomizing the order may be an option? With the 24 test instances we currently have that would significantly lower the probability.

Reply:
Yes, indeed. My thought was to prevent booking an instance that's already used in another PR. So I used @lithrel's hint and reversed the order. If we have multiple PRs that with no tickets, then there is not much we can do.

One other approach would be to randomize  the list 🤔

Reply:
Looks like our comments also experienced a race condition :smile:

Reply:
Commit amended. It randomizes the list if it's a case of no ticket.